### PR TITLE
Changed API of comparison/sort/enumeration/qos options to match Darwin version

### PR DESCRIPTION
**Pull request by tomquist**
Changed API of NSComparisonResult, NSQualityOfService, NSSortOptions and NSEnumerationOptions to match Darwin version

### CODE REVIEW

File: Foundation/NSOperation.swift [hunk 1]
```diff
@@ -141,7 +141,7 @@ public class NSOperation : NSObject {
     public var threadPriority: Double = 0.5
     
     /// - Note: Quality of service is not directly supported here since there are not qos class promotions available outside of darwin targets.
-    public var qualityOfService: NSQualityOfService = .Default
+    public var qualityOfService: NSQualityOfService = .`default`
```
Comment (parkera) [inline Foundation/NSOperation.swift]: Hm, I thought an enhancement to the compiler was going to fix this ugly backtick case for enum values.

**Comment by parkera** [comment]
@swift-ci please test

File: Foundation/NSOperation.swift (hunk 1 above)
Comment (tomquist) [inline Foundation/NSOperation.swift]: I suppose you are right. I'll check that.

**Comment by parkera** [comment]
Possibly missed case?

```
/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift-corelibs-xctest/Sources/XCTest/XCTestCase.swift:287:50: error: type 'NSComparisonResult' has no member 'OrderedAscending'
        } while NSDate().compare(timeoutDate) == NSComparisonResult.OrderedAscending
                                                 ^~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~
```

**Comment by parkera** [comment]
Oh I see, it's in XCTest. We'll have to make a corresponding PR for them.

File: Foundation/NSOperation.swift (hunk 1 above)
Comment (tomquist) [inline Foundation/NSOperation.swift]: This has been fixed but the enum case declaration still requires backticks.

**Pull request merged**